xarray/core/indexing.py

```diff
@@ -902,18 +906,18 @@ def _outer_to_vectorized_indexer(
 
     n_dim = len([k for k in key if not isinstance(k, integer_types)])
     i_dim = 0
-    new_key = []
+    new_key = ()
     for k, size in zip(key, shape):
         if isinstance(k, integer_types):
-            new_key.append(np.array(k).reshape((1,) * n_dim))
+            new_key += (np.array(k).reshape((1,) * n_dim),)
```

Comment:
It sounds like my prior is wrong, but am surprised tuples are faster than lists here — isn't python copying the whole tuple on each append, given tuples are immutable?

Reply:
not anymore. since like 3.9 or something there are some tuple appending optimization in cpython where the tuple object is now recycled for some of these common paths.

Reply:
Cool! Thanks, TIL